### PR TITLE
Override bumblebee download view.

### LINKED ISSUE #2131: bumblebee_download delivers wrong file for checked out documents.
With https://github.com/4teamwork/opengever.core/pull/1900 we disabled the checksum update for checked out documents. 

When storing checked out documents in bumblebee for the first time the `bumblebee_download` delivers the current file which will result in a checksum mismatch in the bumblebee service. We should override `bumblebee_download` to deliver the correct file for the registered checksum of checked out files (probably the most recent version).

### PR DESCRIPTION
**Pull request by deiferni**
Fixes an issue with activating bumblebee when documents are checked out
during the process of activation.

Fixes #2131.

### CODE REVIEW

File: opengever/bumblebee/tests/test_download.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+from zope.event import notify
+from zope.lifecycleevent import ObjectModifiedEvent
+import transaction
+
+
+class TestBumblebeeDownloadView(FunctionalTestCase):
+    """Test the bumblebee_download view.
+
+    Duplicates methods from ftw.bumblebee tests since we still want the
+    7original behaviour as long as the document is not checked out.
```
Comment (phgross) [inline opengever/bumblebee/tests/test_download.py]: **7**original

File: opengever/bumblebee/tests/test_download.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+from zExceptions import NotFound
+from zope.event import notify
+from zope.lifecycleevent import ObjectModifiedEvent
+import transaction
+
+
+class TestBumblebeeDownloadView(FunctionalTestCase):
+    """Test the bumblebee_download view.
+
+    Duplicates methods from ftw.bumblebee tests since we still want the
```
Comment (phgross) [inline opengever/bumblebee/tests/test_download.py]: Why duplicating the tests, doesn't inheriting work ?

**Comment by deiferni** [comment]
@phgross plz have a look

**Comment by deiferni** [comment]
@phgross updated

File: opengever/bumblebee/tests/test_download.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                     .within(self.dossier))
+        initial_checksum = IBumblebeeDocument(doc).get_checksum()
+
+        # We store these parameters as they were sent to bumblebee with /store
+        params = {'token': download_token_for(doc),
+                  'uuid': IUUID(doc),
+                  'checksum': initial_checksum}
+
+        # then the file changes, a new version is created.
+        set_file(doc, "A Freaking Mail", filename=u"example.eml")
```
Comment (phgross) [inline opengever/bumblebee/tests/test_download.py]: Don't use *.eml that should never happen in GEVER.

File: opengever/bumblebee/tests/test_download.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        doc.setModificationDate(datetime.now())
+        notify(ObjectEditedEvent(doc))  # AT only.
```
Comment (phgross) [inline opengever/bumblebee/tests/test_download.py]: #AT only ?

**Comment by deiferni** [comment]
@phgross updated

**Pull request merged**